lib/vimgolf/cli.rb

```diff
@@ -61,43 +60,69 @@ def setup
       end
     end
 
-    desc "put [ID]", "launch Vim session"
+    desc "put CHALLENGE_ID", "launch vimgolf.com challenge"
     long_desc <<-DESC
     Launch a VimGolf session for the specified challenge ID. To find a currently
     active challenge ID, please visit vimgolf.com!
     DESC
 
-    def put(id = nil)
+    def put(id)
       FileUtils.mkdir_p Config.put_path
       VimGolf.ui.warn "Downloading Vimgolf challenge: #{id}"
       VimGolf::Challenge.path(Config.put_path)
       challenge = Challenge.new(id)
       challenge.download
 
+      play(challenge)
+    end
+
+    desc "local INFILE OUTFILE", "launch local challenge"
+    long_desc <<-DESC
+    Launch a local VimGolf challenge. A temporary copy of INFILE is made; the original files will not be touched.
+    DESC
+
+    def local(infile, outfile)
+      # make sure our files are sane
+      if !(File.file?(infile) and File.file?(outfile))
+        VimGolf.ui.error "INFILE and OUTFILE must exist and be regular files."
+        exit 1
+      end
+
+      challenge = Challenge.new(infile) # use the filename as id
```

Comment:
I mean.. challenge.id already contains the ID, so passing infile to local() seems unnecessary, since you have the value initialized from the constructor.